lib/sqlalchemy/sql/functions.py

```diff
@@ -449,7 +452,21 @@ def within_group(self, *order_by):
         """
         return WithinGroup(self, *order_by)
 
-    def filter(self, *criterion):
+    @overload
+    def filter(self) -> Self:
+        ...
+
+    @overload
+    def filter(
+        self,
+        __criterion0: _ColumnExpressionArgument[bool],
+        *criterion: _ColumnExpressionArgument[bool],
```

Comment:
**Federico Caselli** (CaselIT) wrote:

if this is 2.1 one we could use `/` to avoid the __ convention

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4972

Reply:
**Michael Bayer** (zzzeek) wrote:

let's do it in a subsequent commit, want to keep a clean cherry pick for the moment

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4972